### DISCUSSION TITLE
A few features / options during check out flow & modify order statuses

### DISCUSSION THREAD

**Discussion by jxs714** [pretix/pretix#4479] [Ideas]
Hey guys, Just feedback. Would be great to have the following:

**Asking questions for overall order before checkout**
Being able to ask questions prior to users finalizing payment. For example, we are using pretix signature option and would like for the user making payment to sign their signature for the entire order, rather than having them do it per ticket attendee. I know you guys dont support the signature plugin, but I think this feature can be useful for other users for specific overall order question. Maybe an alternative to the existing "Terms and Conditions" check box option that is more customizable and can add multiple questions.

**Enabling/Disabling answers to question after order**
We are using the pretix option to allow changes to all order information. Would be great if there was an option in questions (and custom questions) to disable changing answers to for specific questions after an order. Maybe the option to only allow changing questions such as transferring attendee info

For example, in our case we have the signature pretix plugin and some users did not sign their signature for their order and somehow skipped it. We now have them returning to their order just to sign their signature, which they can change again or delete at anytime after we confirm with them via email since the option to allow changes to the order is enabled. Maybe also an option to lock changes to a specific order would be great as well. At this moment disabling the option to modify order details would apply to all orders, which means that users will not be able to add their signature / update questions. 

**Change a ticket to a pending, rather than cancel when ticket is missing information**
When canceling an order (prior to initiating refund) we can send a message to customers before refunding. Doing this stops the customer from editing any information on their ticket(s). For us in our case, we need their signature for their order to re-activate their ticket. They can't do this / update their ticket details if the order is cancelled. Would be great if we can have an option to change orders from PAID / ACTIVE to Pending, with the same options as Canceling but with the ability for customers to change their answers to their questions for the order until admin can approve and change the status back to Active / Paid.

**Customizing what information to show on order details**
An option to enable and see data for any questions, information on customer or internal notes in the orders section so that we can view everything on a customer order that we need as admin. Right now we can only see order #, user, order date, total, positions and status. An option to also filter the results with internal notes and others.

In my case, id have to search for the keywords in "internal comments" for each order that I saved the internal comment in order for me to find them. 

Hope this helps and maybe these options exists. Thanks guys

**Comment by raphaelm** [comment]
> Asking questions for overall order before checkout
> Being able to ask questions prior to users finalizing payment. For example, we are using pretix signature option and would like for the user making payment to sign their signature for the entire order, rather than having them do it per ticket attendee. I know you guys dont support the signature plugin, but I think this feature can be useful for other users for specific overall order question. Maybe an alternative to the existing "Terms and Conditions" check box option that is more customizable and can add multiple questions.

This is a common request. Currently we do not plan on adding it because it adds a lot of complexity to both frontend and backend, but we might re-evaluate in the future.

> Enabling/Disabling answers to question after order
> We are using the pretix option to allow changes to all order information. Would be great if there was an option in questions (and custom questions) to disable changing answers to for specific questions after an order. Maybe the option to only allow changing questions such as transferring attendee info

I believe this will also quickly get very complex, especially given that questions can depend on each other.

> Change a ticket to a pending, rather than cancel when ticket is missing information
> When canceling an order (prior to initiating refund) we can send a message to customers before refunding. Doing this stops the customer from editing any information on their ticket(s). For us in our case, we need their signature for their order to re-activate their ticket. They can't do this / update their ticket details if the order is cancelled. Would be great if we can have an option to change orders from PAID / ACTIVE to Pending, with the same options as Canceling but with the ability for customers to change their answers to their questions for the order until admin can approve and change the status back to Active / Paid

This sounds like quite a specific workflow, I'm not sure we'll be able to support it, maybe some time in the future.

> Customizing what information to show on order details
> An option to enable and see data for any questions, information on customer or internal notes in the orders section so that we can view everything on a customer order that we need as admin. Right now we can only see order #, user, order date, total, positions and status. An option to also filter the results with internal notes and others.

The orders section is a list of orders, not order positions, therefore we can't show things like questions.
